### PR TITLE
Use assemleTaskProvider directly

### LINKED ISSUE #120: set assembleProvider dependency using the `configure` method
`variant.assembleProvider.configure { it.finalizedBy(uploadNativeSymbolsTask) }`

inspired by https://github.com/remcomokveld/sentry-android-gradle-plugin/commit/67fcdd8ef3e7bec64537f631876ab3bfd614e489
so we don't have to `assembleProvider.get()`

### PR DESCRIPTION
**Pull request by cortinico**
## :scroll: Description
I've switched to use `assemleTaskProvider` directly without calling `.get()` on it. 
Plus I refactored the `withLogging` into a Util class.

## :bulb: Motivation and Context
Fixes #120 

## :green_heart: How did you test it?
Haven't touched the tests, they're green locally 👍 

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] No breaking changes

### CODE REVIEW

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt [hunk 1]
```diff
@@ -166,7 +162,7 @@ class SentryPlugin : Plugin<Project> {
                     // uploadNativeSymbolsTask will only be executed after the assemble task
                     // and also only if `uploadNativeSymbols` is enabled, as this is an opt-in feature.
                     if (extension.uploadNativeSymbols.get()) {
-                        assembleTask?.finalizedBy(uploadNativeSymbolsTask)
+                        assembleTask?.configure { it.finalizedBy(uploadNativeSymbolsTask) }
```
Comment (marandaneto) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt:165]:
```suggestion
                        variant.assembleProvider.configure { it.finalizedBy(uploadNativeSymbolsTask) }
```

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryUploadProguardMappingsTask.kt [hunk 2]
```diff
@@ -8,12 +8,16 @@ import org.gradle.api.file.RegularFile
...
+import org.gradle.api.tasks.CacheableTask
 import org.gradle.api.tasks.Exec
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputFile
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
 
+@CacheableTask
```
Comment (marandaneto) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryUploadProguardMappingsTask.kt:20]: should this be a separate PR to fix https://github.com/getsentry/sentry-android-gradle-plugin/issues/117 ?

Comment (cortinico) [reply plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryUploadProguardMappingsTask.kt:20]: oops I actually didn't want to push this. I was checking the Gradle warnings raised by this and forgot to clean my working dir. I'm amending this.

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt [hunk 3]
```diff
@@ -55,12 +51,12 @@ class SentryPlugin : Plugin<Project> {
...
 
-                val bundleTask = withLogging("bundleTask") {
+                val bundleTask = withLogging(project.logger, "bundleTask") {
                     getBundleTask(project, variant.name)
                 }
 
-                val assembleTask = withLogging("assembleTask") {
-                    getAssembleTask(variant)
+                val assembleTask = withLoggingProvider(project.logger, "assembleTask") {
+                    getAssembleTaskProvider(variant)
```
[the remark this answers is not in the document]
Comment (cortinico) [reply plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt:59]: 👍 
I thought we wanted to save the logging. If we're fine losing it, I can clean it up

[the remark this answers is not in the document]
Comment (marandaneto) [reply plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt]: it was more a problem when loading it dynamically by task name, but with the provider we are gold, thanks

**Review by marandaneto — APPROVED** [review]
LGTM

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt [hunk 4]
```diff
@@ -166,7 +156,11 @@ class SentryPlugin : Plugin<Project> {
                     // uploadNativeSymbolsTask will only be executed after the assemble task
                     // and also only if `uploadNativeSymbols` is enabled, as this is an opt-in feature.
                     if (extension.uploadNativeSymbols.get()) {
-                        assembleTask?.finalizedBy(uploadNativeSymbolsTask)
+                        variant.assembleProvider?.configure {
```
Comment (marandaneto) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt:159]: getAssembleTaskProvider has been changed but here calls the provider directly
